File: app/src/main/java/org/antrack/app/OSignal.java
package org.antrack.app;

import com.onesignal.OneSignal;

import org.antrack.app.libs.Files;
import org.antrack.app.libs.L;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;

public class OSignal {
    public static void writeId() {
        OneSignal.idsAvailable(new OneSignal.IdsAvailableHandler() {
            @Override
            public void idsAvailable(String userId, String registrationId) {
                try {
                    Files.writeTextFile(Init.MAIN_DIR + C.OSID_FILE, userId);
                } catch (IOException e) {
                    L.e("OSignal", "error: " + e);
                }
            }
        });
    }

    public static void push(String userId) {
        L.d("OSignal", "push to " + userId);

        try {
            OneSignal.postNotification(new JSONObject("{'contents': {'en':'ping'}, 'include_player_ids': ['" + userId + "'], 'priority':10}"),
                    new OneSignal.PostNotificationResponseHandler() {
                        @Override
                        public void onSuccess(JSONObject response) {
                            L.d("OSignal", "postNotification Success: " + response.toString());
                        }

                        @Override
                        public void onFailure(JSONObject response) {
                            L.e("OSignal", "postNotification Failure: " + response.toString());
                        }
                    });
        } catch (JSONException e) {
            e.printStackTrace();
        }
    }
}

File: app/src/main/java/org/antrack/app/OSignal.kt
package org.antrack.app

import com.onesignal.OneSignal

import org.antrack.app.libs.Files
import org.antrack.app.libs.L
import org.json.JSONException
import org.json.JSONObject

import java.io.IOException

object OSignal {
    fun writeId() {
        OneSignal.idsAvailable { userId, _ ->
            try {
                Files.writeTextFile(Init.MAIN_DIR + C.OSID_FILE, userId)
            } catch (e: IOException) {
                L.e("OSignal", "error: " + e)
            }
        }
    }

    fun push(userId: String, message: String) {
        L.d("OSignal", "push to " + userId)

        try {
            OneSignal.postNotification(JSONObject("{'contents': {'en':'$message'}, 'include_player_ids': ['$userId'], 'priority':10}"),
                    object : OneSignal.PostNotificationResponseHandler {
                        override fun onSuccess(response: JSONObject) {
                            L.d("OSignal", "postNotification Success: " + response.toString())
                        }

                        override fun onFailure(response: JSONObject) {
                            L.e("OSignal", "postNotification Failure: " + response.toString())
                        }
                    })
        } catch (e: JSONException) {
            e.printStackTrace()
        }
    }
}
